### PR TITLE
Upgrade terraform-plugin-sdk to v2

### PR DESCRIPTION
**Pull request by shoekstra**

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -5,24 +5,50 @@ cloud.google.com/go v0.44.1/go.mod h1:iSa0KzasP4Uvy3f1mN/7PiObzGgflwredwwASm/v6A
 cloud.google.com/go v0.44.2/go.mod h1:60680Gw3Yr4ikxnPRS/oxxkBccT6SA1yMk63TGekxKY=
```
Comment (svanharmelen) [inline go.sum:5]: Did you run `go mod tidy`?

File: mcaf/provider.go [hunk 2]
```diff
@@ -4,22 +4,21 @@ package mcaf
...
 
-	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
-	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-// Provider returns a terraform.ResourceProvider.
-func Provider() terraform.ResourceProvider {
+// New returns a schema.Provider.
+func New() *schema.Provider {
```
Comment (svanharmelen) [inline mcaf/provider.go:11]: Is this a new best practice? While in Go term this is a good change, in TF terms this was the standard way to name this function.

File: go.sum (hunk 1 above)
Comment (shoekstra) [reply go.sum:5]: I did, just ran again and no changes.

File: mcaf/provider.go (hunk 2 above)
Comment (shoekstra) [reply mcaf/provider.go:11]: Seems so, noticed the change when checking to out the [terraform-provider-scaffolding](https://github.com/hashicorp/terraform-provider-scaffolding) project. Felt like a good time to align with that.

Wasn't so sure on moving `provider` to `internal/provider` though.

Comment (svanharmelen) [reply mcaf/provider.go:11]: Check... Well I should move it to `internal` as well. That is a good pattern to adopt I think...

**Review by svanharmelen — APPROVED** [review]

**Pull request merged**